osu.Game/Screens/Tournament/Components/VisualiserLine.cs

```diff
@@ -0,0 +1,209 @@
+using osu.Framework.Graphics.Primitives;
+using osu.Framework.Timing;
+using osu.Framework.Graphics.Transforms;
+
+namespace osu.Game.Screens.Tournament.Components
+{
+    class VisualiserLine : Drawable
+    {
+        private float strokeWidth = 1;
+        public float StrokeWidth
+        {
+            get { return strokeWidth; }
+            set
+            {
+                if (strokeWidth == value)
+                    return;
+
+                strokeWidth = value;
+                Invalidate(Invalidation.DrawNode, shallPropagate: false);
+            }
+        }
+
+        private float strokeHeight = 1;
+        public float StrokeHeight
+        {
+            get { return strokeHeight; }
+            set
+            {
+                if (strokeHeight == value)
+                    return;
+
+                strokeHeight = value;
+                Invalidate(Invalidation.DrawNode, shallPropagate: false);
+            }
+        }
+
+        private float separation = 0;
+        public float Separation
+        {
+            get { return separation; }
+            set
+            {
+                if (separation == value)
+                    return;
+
+                separation = value;
+                Invalidate(Invalidation.DrawNode, shallPropagate: false);
+            }
+        }
+
+        private Shader shader;
+
+        private VisualiserLineDrawNodeSharedData visualiserLineDrawNodeSharedData => new VisualiserLineDrawNodeSharedData();
+
+        /// <summary>
+        /// The period of this visualiser line, in radians.
+        /// </summary>
+        private float period;
+
+        /// <summary>
+        /// The period offset this line was constructed with, in radians.
+        /// </summary>
+        private float initialPeriodOffset;
+        /// <summary>
+        /// The rolling period offset (by transformation), in radians.
+        /// </summary>
+        private float internalPeriodOffset;
+        /// <summary>
+        /// The final period offset, in radians.
+        /// </summary>
+        private float periodOffset
+        {
+            get { return initialPeriodOffset + internalPeriodOffset; }
+            set
+            {
+                if (internalPeriodOffset == value)
+                    return;
+
+                internalPeriodOffset = value;
+                Invalidate(Invalidation.DrawNode, shallPropagate: false);
+            }
+        }
+
+        /// <summary>
+        /// Constructs a new Visualiser Line.
+        /// </summary>
+        /// <param name="period">The period of the line, in radians.</param>
+        /// <param name="periodOffset">The offset to the period of the line, in radians.</param>
+        /// <param name="cycleTime">The time to cycle the line.</param>
+        public VisualiserLine(float period, float periodOffset = 0, int cycleTime = 0)
+        {
+            this.period = period;
+            this.initialPeriodOffset = periodOffset;

```

Comment:
I disagree, initialPeriodOffset is different from periodOffset and should actually remain unchanged at all times (will add a readonly).